File: src/main/java/com/ByTech/ByTech/Cart/services/DetailService.java
package com.ByTech.ByTech.Cart.services;

import com.ByTech.ByTech.Cart.models.Detail;
import com.ByTech.ByTech.Cart.repositories.DetailRepository;
import jakarta.transaction.Transactional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
@Transactional
public class DetailService {

    private final DetailRepository detailRepository;

    @Autowired
    public DetailService(DetailRepository detailRepository) {
        this.detailRepository = detailRepository;
    }

    public void createDetail(Detail detail) {
        this.detailRepository.save(detail);
    }

    public List<Detail> getDetailBySale(Long saleId) {
        return this.detailRepository.findBySale_Id(saleId);
    }
}

File: Movil/backend/ByTech_movil/src/main/kotlin/com/example/ByTech_movil/Detail/services/DetailService.kt
package com.example.ByTech_movil.Detail.services

import com.example.ByTech_movil.Detail.models.Detail
import com.example.ByTech_movil.Detail.repositories.DetailRepository
import jakarta.transaction.Transactional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
@Transactional
class DetailService @Autowired constructor(private val detailRepository: DetailRepository) {

    fun createDetail(detail: Detail?) {
        detail?.let { detailRepository.save(it) }
    }

    fun getDetailBySale(saleId: Long?): List<Detail> {
        return detailRepository.findBySale_Id(saleId)
    }
}
